rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,125 @@
+
+# Guide-level explanation
+
+GTFS output includes additional, hardcoded data about where rail lines go to and where they can stop
+if all stations were open; these hardcoded data are independent of all scheduled service except that
+they may be tied to scheduled route patterns in some cases.
+
+This is represented as a service with ID "canonical", which is valid for the entire feed date range,
+but which has no active days of the week. The typicality of the service is tagged with a new value
+so it can be programmatically identified.
+
+All rail routes have two or more trips on this service, corresponding to the "canonical" patterns of
+service in each direction for any branches, as understood by stakeholders. Each such canonical trip
+has a route pattern, for which it is the representative trip, and a shape. If current service has
+trips following the same set and order of stops as the canonical trip, then the canonical trip will
+reference—and become the representative trip for—the current service's route pattern. If current
+service has no trips resembling the canonical trip, then a separate, canonical route pattern will be
+maintained in the GTFS output.
+
+# Reference-level explanation
+
+The canonical service appears in the following GTFS files in the following way:
+
+- **calendar.txt** includes a new `service_id` to envelop all of the canonical trips on all the
+  routes. `start_date` and `end_date` should reflect the feed's valid start/end dates. All other
+  fields should have a value of `0`.
+- **calendar_attributes.txt** includes the new `service_id`. `service_schedule_type` should be
+  `"Other"`, and `rating_start_date` and `rating_end_date` should reflect the feed's valid start/end
+  dates. `service_schedule_typicality` should be `6`, and the
+  [docs](https://github.com/mbta/gtfs-documentation/blob/master/reference/gtfs.md) should be updated
+  with this new kind of value.
+- **route_patterns.txt** includes new route patterns for routes/directions we want to hardcode when
+  the canonical trips (added in the trips file below) do not resemble any scheduled service. New
+  patterns in this file shall have a new `route_pattern_typicality` value of should be `6`, and
+  `route_pattern_sort_order` should be a greater than the value for all non-canonical route patterns
+  for that given route/direction/branch. This will be done by adding two new digits to
+  `route_pattern_sort_order` (one for branch and the other for canonicalness, both placed right
+  after the existing digits corresponding to the `route_sort_order`). For routes with branches,
+  you'll need a pattern for each route/direction/branch (eg, for Providence/Stoughton Line, you
+  should have _two_ new route patterns in the outbound direction). Finally, a new
+  `canonical_route_pattern` field is added to all route patterns, indicating whether or not it can
+  be considered canonical (or if the route does not have canonical patterns defined, as will be the
```

Comment:
We should specify values for this field. I think there was discussion elsewhere about this being tri-state, i.e. values for "this is canonical", "this isn't canonical", and "no comment" (for routes / route types we don't apply this system to).

Reply:
Good idea. 👍